cmd/policygen/main.go

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// policygen automates generation of Google-recommended Policy Library constraints based on your Terraform configs.
+//
+// Usage:
+// $ policygen --input_config=./samples/config.yaml --output_dir=/tmp/constraints {--input_dir=/path/to/configs/dir|--input_plan=/path/to/plan/json|--input_state=/path/to/state/json}
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"flag"
+
+	"github.com/GoogleCloudPlatform/healthcare-data-protection-suite/internal/pathutil"
+	"github.com/GoogleCloudPlatform/healthcare-data-protection-suite/internal/runner"
+	"github.com/GoogleCloudPlatform/healthcare-data-protection-suite/internal/template"
+	"github.com/GoogleCloudPlatform/healthcare-data-protection-suite/internal/terraform"
+	"github.com/ghodss/yaml"
+)
+
+var (
+	inputConfig = flag.String("input_config", "", "Path to the Policy Generator config.")
+	inputDir    = flag.String("input_dir", "", "Path to Terraform configs root directory. Cannot be specified together with other types of inputs.")
+	inputPlan   = flag.String("input_plan", "", "Path to Terraform plan in json format, Cannot be specified together with other types of inputs.")
+	inputState  = flag.String("input_state", "", "Path to Terraform state in json format. Cannot be specified together with other types of inputs.")
+	outputDir   = flag.String("output_dir", "", "Path to directory to write generated policies")
+)
+
+// Config is the struct representing the Policy Generator configuration.
+type Config struct {
+	OrgID           string                 `json:"org_id"`
+	ForsetiPolicies map[string]interface{} `json:"forseti_policies"`
+	GCPOrgPolicies  map[string]interface{} `json:"gcp_organization_policies"`
+}
+
+// TODO: Move this to input config.
```

Comment:
We should still try to assign to a person or issue if we can

Reply:
Done.